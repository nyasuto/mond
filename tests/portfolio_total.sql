INSERT INTO assets (ticker, ccy) VALUES ('VTI','USD');
INSERT INTO assets (ticker, ccy) VALUES ('TOPIX','JPY');

INSERT INTO fx_rates (date, pair, rate) VALUES ('2023-12-29','USDJPY',140.0);
INSERT INTO fx_rates (date, pair, rate) VALUES ('2023-12-30','USDJPY',142.0);

INSERT INTO snapshots (date, ticker, qty, price_ccy) VALUES ('2023-12-29','VTI',10,200);
INSERT INTO snapshots (date, ticker, qty, price_ccy) VALUES ('2023-12-29','TOPIX',5,100);
INSERT INTO snapshots (date, ticker, qty, price_ccy) VALUES ('2023-12-30','VTI',11,205);
INSERT INTO snapshots (date, ticker, qty, price_ccy) VALUES ('2023-12-30','TOPIX',5,101);

SELECT
  date,
  round(total_value_jpy, 3) AS total_value_jpy
FROM v_portfolio_total
ORDER BY date;
