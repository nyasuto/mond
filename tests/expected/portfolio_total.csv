date,total_value_jpy
2023-12-29,280500.0
2023-12-30,320715.0
